synapse/rest/client/mutual_rooms.py

```diff
@@ -67,8 +56,8 @@ async def on_GET(
                 errcode=Codes.FORBIDDEN,
             )
 
-        rooms = await self.store.get_mutual_rooms_for_users(
-            requester.user.to_string(), user_id
+        rooms = await self.store.get_mutual_rooms_between_users(
+            frozenset((requester.user.to_string(), user_id))
```

Comment:
Why the `frozenset` cast? Doesn't this already return a `frozenset`?

Reply:
This requires a frozen set for arguments to make it cache-able, `set` isn’t hashable.

Reply:
Right, I read this as casting the result for some reason.